### PR TITLE
postgresql@95: import from homebrew/versions.

### PR DESCRIPTION
**Pull request by alanorth**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Import PostgreSQL 9.5.5 from homebrew/versions. After that we can bump version to 9.5.6 and close the [corresponding version bump in homebrew/versions](https://github.com/Homebrew/homebrew-versions/pull/1533).

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Could you also import `postgresql94.rb` too? Thanks!

File: Formula/postgresql@95.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
+class PostgresqlAT95 < Formula
+  desc "Object-relational database system"
+  homepage "https://www.postgresql.org/"
+  url "https://ftp.postgresql.org/pub/source/v9.5.5/postgresql-9.5.5.tar.bz2"
```
Comment (SrihariThalla) [inline Formula/postgresql@95.rb]: Can you update to `9.5.6`? Thanks!

Comment (alanorth) [inline Formula/postgresql@95.rb]: I was [planning on updating to 9.5.6](https://github.com/Homebrew/homebrew-versions/pull/1533) when I saw that homebrew-versions is deprecated. I was going move postgresql95 here first, then update after. What do you think, @MikeMcQuaid?

**Comment by alanorth** [comment]
@MikeMcQuaid I imported `postgresql94.rb` as `postgresql@94.rb` as you requested.

File: Formula/postgresql@95.rb (hunk 1 above)
Comment (MikeMcQuaid) [inline Formula/postgresql@95.rb]: Ideally it's worth just doing both in a single PR, I think, but feel free to 👎 that.

Comment (alanorth) [inline Formula/postgresql@95.rb]: I've just bumped the version of `postgresql@95` to 9.5.6 and added it to this pull request.

**Comment by MikeMcQuaid** [comment]
Some CI failures: https://bot.brew.sh/job/Homebrew%20Core/17569/version=sierra/testReport/

**Review by alanorth — COMMENTED** [review]
@JCount the post-installation help now gives this cryptic message:

```
To have launchd start postgresql@95 now and restart at login:
  brew services start postgresql@95
Or, if you don't want/need a background service you can just run:
  pg_ctl -D /opt/brew/var/Formulary::FormulaNamespaceed575915b5ab63d58c0ac7d92c6c6b95::PostgresqlAT95 start
```

File: Formula/postgresql@95.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    To migrate existing data from a previous minor version (9.0-9.4) of PostgreSQL, see:
+      https://www.postgresql.org/docs/9.5/static/pgupgrade.html
+
+      You will need your previous PostgreSQL installation from brew to perform `pg_upgrade`.
+      Do not run `brew cleanup postgresql@95` until you have performed the migration.
+    EOS
+  end
+
+  plist_options :manual => "pg_ctl -D #{HOMEBREW_PREFIX}/var/#{name} start"
```
Comment (MikeMcQuaid) [inline Formula/postgresql@95.rb]: Don't use `name` here; just hardcode it.

File: Formula/postgresql@94.rb [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  depends_on "readline"
+  depends_on "libxml2" if MacOS.version <= :leopard # Leopard libxml is too old
+  depends_on :python => :optional
+
+  fails_with :clang do
+    build 211
+    cause "Miscompilation resulting in segfault on queries"
+  end
+
+  patch :DATA
```
Comment (MikeMcQuaid) [inline Formula/postgresql@94.rb]: I think `ENV.append_to_cflags "-D_XOPEN_SOURCE"` should work instead.

Comment (MikeMcQuaid) [inline Formula/postgresql@94.rb]: (also, while you're making that change, the same thing on `postgres-xc` would be great if you've got a chance!)

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by JCount** [comment]
This version of `postgresql@95` that I pushed to the repository was basically me taking an ax and butchering the code until it would compile and "work" as a `keg_only :versioned_formula`. I myself have never actually used the software for anything, so any testing/feedback from people who actually do would be extremely helpful.

File: Formula/postgresql@95.rb (hunk 2 above)
Comment (JCount) [inline Formula/postgresql@95.rb]: Should I hardcode it within the plist itself as well?

**Comment by alanorth** [comment]
@JCount and initially I was only porting it from homebrew/versions, I didn't know it would be such a complicated change! I have no idea what it needs now. 😄

**Comment by JCount** [comment]
@alanorth I think it is getting close to being done with the addition of these latest commits I pushed. 😌

**Comment by JCount** [comment]
@MikeMcQuaid Do you want to look this over? If you think it is good, we probably can get this merged.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by JCount** [comment]
The forumla names need to be changed to `postgresql@9.4` and `postgresql@9.5`, and the formula contents updated to reflect this, before this PR can be merged.

**Review by JCount — CHANGES_REQUESTED** [review]
The forumla names need to be changed to postgresql@9.4 and postgresql@9.5, and the formula contents updated to reflect this, before this PR can be merged.

**Review by JCount — APPROVED** [review]

**Comment by JCount** [comment]
@alanorth Thanks a lot for you contribution to Homebrew, and especially your patience in getting this merged. 🎊

**Pull request closed without merging**